test/functional/tests/failover_standby/test_cache_activation_perf.py

```diff
@@ -0,0 +1,151 @@
+
+runtime = timedelta(days=1)
+activation_time_threshold = timedelta(seconds=10)
+
+test_drive_size = Size(800, Unit.GibiByte)
+
+
+def get_fio_rw_cmd(core, write_percentage):
+    fio = (
+        Fio()
+        .create_command()
+        .target(core)
+        .read_write(ReadWrite.randrw)
+        .write_percentage(write_percentage)
+        .io_engine(IoEngine.libaio)
+        .block_size(Size(16, Unit.Blocks4096))
+        .run_time(runtime)
+        .time_based(runtime)
+        .io_depth(32)
+        .num_jobs(72)
+        .direct(1)
+    )
+    return fio
+
+
+@pytest.mark.require_disk("cache_1", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("cache_2", DiskTypeSet([DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeSet([DiskType.hdd]))
+def test_cache_activation_time():
+    """
+    title: Measure standby cache activation time
+    description: Determine if the activation of a standby cache completes within
+                 a predefined time limit.
+    pass_criteria:
+      - The activation completes within the time limit
+    """
+    with TestRun.step("Setting up two cache devices, 800GB each"):
+        cache_devices = [TestRun.disks['cache_1'], TestRun.disks['cache_2']]
+        core_device = TestRun.disks['core']
+
+        # cache devices - create partitions
+        for cache_dev in cache_devices:
+            cache_dev.create_partitions([test_drive_size])
+
+        # core partition creation
+        core_device.create_partitions([test_drive_size])
+
+    with TestRun.step("Start cache and add core"):
+        cache = casadm.start_cache(cache_devices[0],
+                                   force=True,
+                                   cache_mode=CacheMode.WB,
+                                   cache_line_size=CacheLineSize.LINE_4KiB)
+
+        cache.add_core(core_device)
+        current_cache_id = cache.cache_id
+
+    with TestRun.step("Disable seq. cutoff, cleaning policy and udev"):
+        Udev.disable()
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step("Fill the cache with dirty data"):
+
+        fill_size = cache.get_statistics().config_stats.cache_size * CacheLineSize.LINE_4KiB
+        (
+            Fio()
+            .create_command()
+            .io_engine(IoEngine.libaio)
+            .block_size(Size(4, Unit.KiB))
+            .read_write(ReadWrite.write)
+            .target(core_device)
+            .size(fill_size)
+            .direct()
+            .run()
+        )
+
+    with TestRun.step("Check usage statistics"):
+        stats = cache.get_statistics(percentage_val=True)
+        if stats.usage_stats.dirty < 98 or stats.usage_stats.occupancy < 98:
+            TestRun.fail(f"Cache is not pre-filled correctly, "
+                         f"dirty={stats.usage_stats_dirty}%, "
+                         f"occupancy={stats.usage_stats.occupancy}%")
```

Comment:
I don't think fail is appropriate here - the stuff we are supposed to test didn't fail - we didn't even do it yet. Maybe `TestRun.LOGGER.error`?

Reply:
fixed